### PR TITLE
Issue-2077 refactor gallery carousel to use tiny-swiper instead of flickity

### PR DESCRIPTION
**Pull request by snovosel**
### Description

refactor the Gallery Carousel block to implement the tiny-swiper package as opposed to Flickity. this removes jQuery as a dependency of this block. 

### Types of changes

code refactor

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -155,7 +155,10 @@
 		"rtlcss-webpack-plugin": "^4.0.3",
 		"simple-git": "^2.35.0",
 		"slick-carousel": "^1.8.1",
-		"uuid": "^8.3.2"
+		"style-loader": "^0.19.1",
```
Comment (olafleur-godaddy) [inline package.json:158]: I don't see where `style-loader` is used?

File: package.json [hunk 2]
```diff
@@ -155,7 +155,10 @@
...
+		"tiny-swiper": "^2.1.2",
+		"uuid": "^8.3.2",
+		"webpack-fix-style-only-entries": "^0.4.0"
```
Comment (olafleur-godaddy) [inline package.json:161]: Same question for that dependency

File: src/blocks/gallery-carousel/edit.js [hunk 3]
```diff
@@ -2,147 +2,46 @@
...
 import classnames from 'classnames';
-import filter from 'lodash/filter';
-import Flickity from 'react-flickity-component';
 import { GalleryCarouselIcon as icon } from '@godaddy-wordpress/coblocks-icons';
+import { useDispatch } from '@wordpress/data';
 
 /**
- * Internal dependencies
- */
+	* Internal dependencies
```
Comment (olafleur-godaddy) [inline src/blocks/gallery-carousel/edit.js:9]: Your spacing looks weird for these comments

File: src/js/coblocks-tinyswiper-initializer.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
+// import jQuery from 'jquery';
```
Comment (olafleur-godaddy) [inline src/js/coblocks-tinyswiper-initializer.js:1]: Yay! :)

File: package.json (hunk 2 above)
Comment (snovosel) [reply package.json]: good catch

File: src/blocks/gallery-carousel/context.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	gutter: 0,
+	gutterMobile: 0,
+} );
+
+const GalleryContextProvider = ( { gutter, gutterMobile, showThumbnails, children, images, isSelected } ) => {
+	const [ selectedImage, setSelectedImage ] = useState( null );
+	const [ captionFocused, setCaptionFocused ] = useState( false );
+
+	const galleryContext = {
+		selectedImage,
```
Comment (jlemay-godaddy) [inline src/blocks/gallery-carousel/context.js:20]: A-Z order

You should run `yarn test:js -- src/blocks/gallery-carousel` and fix all the warnings

File: src/blocks/gallery-carousel/deprecated.js [hunk 6]
```diff
@@ -14,6 +14,7 @@ import metadata from './block.json';
  * WordPress dependencies
  */
 import { getColorClassName, RichText } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
```
Comment (jlemay-godaddy) [inline src/blocks/gallery-carousel/deprecated.js:17]: You'll catch that running the lint:js, but single imports goes before multiple imports (LN16 wil switch with LN17)

File: src/blocks/gallery-carousel/styles/editor.scss [hunk 7]
```diff
@@ -10,6 +10,7 @@
 	}
 
 	.coblocks-gallery--figure {
+		// set the margin top
```
Comment (jlemay-godaddy) [inline src/blocks/gallery-carousel/styles/editor.scss:13]: rm comment

File: src/js/coblocks-tinyswiper-initializer.js (hunk 4 above)
Comment (jlemay-godaddy) [reply src/js/coblocks-tinyswiper-initializer.js:1]: Remove if unused

File: src/components/swiper/style.scss [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+	justify-content: center;
+	position: relative;
+	width: 100px;
+}
+
+.nav-button {
+	&__prev {
+		background-color: var(--wp-admin-theme-color, #007cba);
+		border: none;
+
```
Comment (jlemay-godaddy) [inline src/components/swiper/style.scss:79]: rm empty line

File: src/components/swiper/style.scss [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+		margin: 0 1%;
+		padding: 15px;
+		position: absolute;
+		top: 50%;
+
+		transform: translateY(-50%);
+		width: 62px;
+	}
+
+	&__next {
```
Comment (jlemay-godaddy) [inline src/components/swiper/style.scss:93]: You should group left and right styles that are shared.
```
&__prev,
&__next  {
    border: none;
    padding: 15px;
    ...
}

&__prev {
    left: 0;
}

&__next {
    right: 0;
}
```

**Comment by olafleur-godaddy** [comment]
The block does not seem to deprecate properly from master to your branch

**Comment by olafleur-godaddy** [comment]
Also, when creating a Carousel block, the Upload/Media Library toolbar is hiding my images, which it does not do in the current master branch version : 

Screen Shot 2021-11-16 at 15 44 40

**Comment by snovosel** [comment]
> The block does not seem to deprecate properly from master to your branch

I should have this fixed in my most recent commit

**Comment by olafleur-godaddy** [comment]
Yes, the deprecation now works here :)

**Comment by olafleur-godaddy** [comment]
Looks great to me! A small difference that I see is that in the Editor side, there is no arrows on the side to switch from one image to another one. If that's an intended change, I don't see any other problem with it :)

**Review by olafleur-godaddy — APPROVED** [review]
Looks good to me if you want to merge!

**Pull request merged**